### PR TITLE
blueberry: init at 1.3.9

### PR DESCRIPTION
**Pull request by romildo**
###### Motivation for this change

Add [blueberry](https://github.com/linuxmint/blueberry), a Bluetooth configuration tool.

It is Linux Mint's spin-off of GNOME Bluetooth, which works in all desktop environments.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by symphorien** [comment]
```
(blueberry.py:16628): com.linuxmint.blueberry-WARNING **: 16:05:45.885: Could not retrieve the BT adapter name with 'bt-adapter -i': Command '['timeout', '2s', 'bt-adapter', '-i']' returned non-zero exit status 127.
```
it looks like it depends on bt-adapter, which is not installed on my system

File: pkgs/tools/bluetooth/blueberry/default.nix [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    runHook preInstall
+
+    mkdir -p $out
+    cp -a etc usr/* $out
+
+    find $out -type f -exec sed \
+      -e "s;/usr/bin/pavucontrol;${pavucontrol}/bin/pavucontrol;" \
+      -e "s;/usr/bin/rfkill;${rfkill}/bin/rfkill;" \
+      -e "s;/usr/lib/blueberry;$out/lib/blueberry;" \
+      -e "s;/usr/sbin/rfkill;${rfkill}/bin/rfkill;" \
```
Comment (Thra11) [inline pkgs/tools/bluetooth/blueberry/default.nix:56]: Should this replace `bt-adapter` with `${bluez-tools}/bin/bt-adapter` too?

**Review by Thra11 — CHANGES_REQUESTED** [review]
I tested this by running `nix-shell -p blueberry` and then running `blueberry`. It built ok, and the GUI loaded, but it kept outputting erros like this whenever I tried to do anything:
```
(blueberry.py:7813): com.linuxmint.blueberry-WARNING **: 14:33:19.022: Could not retrieve the BT adapter name with 'bt-adapter -i': Command '['timeout', '2s', 'bt-adapter', '-i']' returned non-zero exit status 127.
timeout: failed to run command ‘bt-adapter’: No such file or directory
```
`bt-adapter` is in the `bluez-tools` package. I don't think putting `bluez-tools` in `buildInputs` is sufficient to make its binaries accessible to blueberry at runtime. It looks like the `bt-adapter` binary is quite central to this application (as opposed to an optional dependency where it might make sense to leave it up to the user to provide it), so it probably should be there. I'm not 100% sure what the correct solution is (manual substitution of absolute path? propagatedBuildInputs?)

**Comment by symphorien** [comment]
propagatedBuildInputs won't do the trick. Either patching the paths, or add it in PATH in the wrapper.

File: pkgs/tools/bluetooth/blueberry/default.nix (hunk 1 above)
Comment (romildo) [reply pkgs/tools/bluetooth/blueberry/default.nix:56]: Done.

**Comment by Thra11** [comment]
It works for me now. Not sure of the nuances, but would `substituteInPlace` be preferable to `sed`?

**Comment by symphorien** [comment]
It does not require escaping, and prints a warning if no replacement is done, iirc.

**Comment by romildo** [comment]
I have replaced `sed` by `substituteInPlace`, but I had to stop using `find` and be explicitly about which files should be edited.

**Review by Thra11 — APPROVED** [review]
Looks good to me.

**Pull request merged**